javascript/javascript1/week2/FullNameFunction.js

```diff
@@ -0,0 +1,22 @@
+function getFullName(firstname, surname, useFormalName, gender) {
+  if (useFormalName === true) {
+    switch (gender) {
+      case "female":
+        return "Lady " + firstname + " " + surname;
+        break;
+      case "male":
+        return "Lord " + firstname + " " + surname;
+        break;
+      default:
+        return firstname + " " + surname;
+    }
+  } else {
+    return firstname + " " + surname;
+  }
+}
+
+const fullName1 = getFullName("Benjamin", "Hughes", true, "male");
+const fullName2 = getFullName("Rebecca", "Hockley", false, "female");
+
+console.log(fullName1);
+console.log(fullName2);
```

Comment:
I think you should also have a check that would let people know they may have forgotten to type in a name :). 
'Remember to consider someone calling the function with an empty string as firstname and lastname.' 
This exercise is forcing you to consider how you design these logic flows and i think practicing this, is super useful going forward!